### PR TITLE
Quit emoji picker on emoji chosen

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#16416](https://github.com/vector-im/element-web/issues/16416)
Peek 2021-02-13 07-41

### CODE REVIEW

File: src/components/views/rooms/MessageComposer.js [hunk 1]
```diff
@@ -149,7 +149,10 @@ const EmojiButton = ({addEmoji}) => {
         const buttonRect = button.current.getBoundingClientRect();
         const EmojiPicker = sdk.getComponent('emojipicker.EmojiPicker');
         contextMenu = <ContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu} catchTab={false}>
-            <EmojiPicker onChoose={addEmoji} showQuickReactions={true} />
+            <EmojiPicker onChoose={(emoji) => {
```
Comment (jryans) [inline src/components/views/rooms/MessageComposer.js:152]: Using an inline function inside JSX like [can be bad for performance](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render), as it means the prop's value change every time, potentially causing the component and its children to re-render unnecessarily.

For a class based component, the typical fix is to define the event handler as a method on the class so it doesn't change. For a hooks-based component like this one, you can use [`useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback) to memoise the handler.

**Review by jryans — CHANGES_REQUESTED** [review]
A small note on the code side below, otherwise seems technically good, but I'm also unsure of the expected functionality, so I'll ask the Design for input.

**Comment by jryans** [comment]
@matrix-org/design: This PR changes the emoji picker to close once you select something. This seems desirable to me personally, but I notice it would then be different from the sticker picker. (At the moment, both the emoji and sticker pickers remain open after you make a selection.)

What's the expected behaviour, and if we agree to change the emoji picker, should the sticker picker change as well?

File: src/components/views/rooms/MessageComposer.js (hunk 1 above)
Comment (SimonBrandner) [reply src/components/views/rooms/MessageComposer.js]: Sorry for that, I didn't realize this could have a performance effect. Should be fixed now

**Comment by SimonBrandner** [comment]
@jryans, could you please re-add the design review. GitHub has removed it when I re-requested your review...

**Review by jryans — APPROVED** [review]
Thanks, looks good code-wise, so now we just need Design input here.

**Comment by SimonBrandner** [comment]
> but I notice it would then be different from the sticker picker

I should be able to put together a similar PR for the sticker picker

**Review by niquewoodhouse — COMMENTED** [review]
I'm slightly unsure about this atm.  On one hand, I would like to approve as I approved the sticker picker closing on selection.  But I'm not 100% sure this is the same thing.

I do query the suggestion that emoji and stickers should be the same.  Just because they're presented in the same way doesn't mean they're the same thing to users when using them as tools to communicate.

Some people, for example, use a sequence of emojis to explain things, and this would make it much harder for them to do that.  Stickers tend to be richer communication tools, so often don't require or play nicely with other stickers to create one message.  

An interaction that (thinking out loud) might be almost as useful, without making things harder for such people, would be closing the emoji picker on sending the message because we can be confident the user has concluded their selections.

_ftr I benchmarked this behaviour this morning against the following to try to identify what might meet most users' expectations._

**Do these pickers close after a selection?**

- Slack: emojis ✅; stickers  N/A (desktop app)
- Discord emojis ✅; stickers ✅ (web)
- Telegram  emojis 🚫;  stickers 🚫 (desktop app)
- Signal 🚫 (desktop app)
- MacOS emojis ✅; stickers N/A
- Whatsapp emojis 🚫;  stickers ✅ (desktop app)
- Microsoft Teams ❓ (no account)
- Messenger emojis 🚫; stickers ✅ (web)

**Comment by SimonBrandner** [comment]
Given what has been said above, it doesn't make sense to keep this around. If someone has convincing arguments for the fact this would be useful let me know, I am happy to reopen

**Pull request closed without merging**